manual/src/tutorials/moduleexamples.etex

```diff
@@ -130,50 +114,54 @@ Signatures are interfaces for structures. A signature specifies
 which components of a structure are accessible from the outside, and
 with which type. It can be used to hide some components of a structure
 (e.g. local function definitions) or export some components with a
-restricted type. For instance, the signature below specifies the three
-priority queue operations "empty", "insert" and "extract", but not the
-auxiliary function "remove_top". Similarly, it makes the "queue" type
+restricted type. For instance, the signature below specifies the
+queue operations "empty", "add", "top" and "pop", but not the
+auxiliary function "make". Similarly, it makes the "queue" type
 abstract (by not providing its actual representation as a concrete type).
+This ensures that users of the "Fifo" module cannot violate data
+structure invariants that operations rely on, such as ``if the front list
+is empty, the rear list is empty too''.
```

Comment:
```suggestion
is empty, the rear list is also empty''.
```

Reply:
I would agree with "is empty as well", or just "is empty", but "is also empty" feels harder to read.

Reply:
"is empty as well" sounds OK to me. I asked a native english speaker who suggested "must also be empty" or "must also be so" as possibilities. Or we could just stick to the original (I had just found it a bit casual).

Reply:
OK for "must also be empty".  Thanks!